packages/circuits/circom/ecdh.circom

```diff
@@ -0,0 +1,38 @@
+pragma circom 2.1.5;
+
+// circomlib imports
+include "./bitify.circom";
+include "./escalarmulany.circom";
+
+// ECDH Is a a template which allows to generate a shared secret
+// from a private key and a public key
+// on the baby jubjub curve
+// It is important that the private key is hashed and pruned first
+// which can be accomplished using the function 
+// deriveScalar from @zk-kit/baby-jubjub
+template Ecdh() {
+    // the private key must pass through deriveScalar first 
+    signal input privateKey;
+    signal input publicKey[2];
+    
+    signal output sharedKey[2];
+
+    // convert the private key to its bits representation
+    component privBits = Num2Bits(253);
+    privBits.in <== privateKey;
+    
+    // multiply the public key by the private key
+    component mulFix = EscalarMulAny(253);
+    mulFix.p[0] <== publicKey[0];
+    mulFix.p[1] <== publicKey[1];
+
+    // need to loop for the length of the private key
+    for (var i = 0; i < 253; i++) {
+        mulFix.e[i] <== privBits.out[i];
+    }
+
+    // we can then wire the output to the shared secret signal
+    sharedKey[0] <== mulFix.out[0];
+    sharedKey[1] <== mulFix.out[1];
+}
```

Comment:
We can use [Anonymous Components](https://docs.circom.io/circom-language/anonymous-components-and-tuples/) here.

```suggestion
template Ecdh() {
    // the private key must pass through deriveScalar first 
    signal input privateKey;
    signal input publicKey[2];
    
    signal output sharedKey[2];
    
    // convert the private key to its bits representation
    var out[253];
    out = Num2Bits(253)(privateKey);

    // multiply the public key by the private key
    var mulFix[2];
    mulFix = EscalarMulAny(253)(out, publicKey);

    // we can then wire the output to the shared secret signal
    sharedKey <== mulFix;
}
```

Reply:
oh so that's how they work 😆 great stuff thank you